commands/init.go

```diff
@@ -39,32 +39,38 @@ func transformInitArgs(args *Args) error {
 		return nil
 	}
 
-	var (
-		name   string
-		newDir bool
-		err    error
-	)
+	var err error
+	dirToInit := "."
+	hasValueRegxp := regexp.MustCompile("^--(template|separate-git-dir|shared)$")
 
-	if args.IsParamsEmpty() {
-		name, err = utils.DirName()
-		if err != nil {
-			return err
+	// Find the first argument that isn't related to any of the init flags.
+	// We assume this is the optional `directory` argument to git init.
+	for i := 0; i < args.ParamsSize(); i++ {
+		arg := args.Params[i]
+		if hasValueRegxp.MatchString(arg) {
+			i++
+		} else if !strings.HasPrefix(arg, "-") {
+			dirToInit = arg
+			break
 		}
-	} else {
-		name = args.LastParam()
-		newDir = true
 	}
 
-	project := github.NewProject("", name, "")
+	dirToInit, err = filepath.Abs(dirToInit)
+	if err != nil {
+		return err
+	}
+
+	// Assume that the name of the working directory is going to be the name of
+	// the project on GitHub.
+	projectName := strings.Replace(filepath.Base(dirToInit), " ", "-", -1)
```

Comment:
Replacing spaces with dashes is what `DirName()` did for us internally, but having that logic here makes it more clear how we construct a github project name. Personally, I found the fact that `DirName()` doesn't return the exact directory name kind of surprising. Also, the code here ensures that the project name is sanitized even if it came from the explicit `directory` argument rather than current working directory.